sklearn/gaussian_process/kernels.py

```diff
@@ -1320,9 +1320,12 @@ def __call__(self, X, Y=None, eval_gradient=False):
                 D = squareform(dists**2)[:, :, np.newaxis]
 
             if self.nu == 0.5:
-                K_gradient = K[..., np.newaxis] * D \
-                    / np.sqrt(D.sum(2))[:, :, np.newaxis]
-                K_gradient[~np.isfinite(K_gradient)] = 0
+                K_gradient = np.zeros_like(D)
```

Comment:
Is this faster or slower than the previous solution? Alternatively we could just catch the warning, right?